examples/demo-rollup/tests/evm/evm_tx.rs

```diff
@@ -55,7 +55,7 @@ async fn sanity_checks(test_client: &SimpleStorageClient) {
 
     assert!(latest_block.number.unwrap().as_u64() > 0);
     assert!(latest_block.number > earliest_block.number);
-    assert!(pending_block.number > latest_block.number);
+    assert!(pending_block.number == latest_block.number);
```

Comment:
Same here. It's not enough to make this test pass. We've changed the semantics and therefore those assertions now don't make sense any more.
We used to have two objects in hand and checked how they relate. Now we have one object that can be accessed in two different ways. So let's first check that both ways of accessing return the same object and then do all assertions on one instance of it.